### ISSUE TITLE
lunari hammer alt impact is sideways

### ISSUE THREAD

**Issue by sayterdarkwynd** [sayterdarkwynd/FrackinUniverse#283]
should spawn on ground facing upwards, not in the air facing sideways

specifically, the fugroundslameice.lua  with the  world.spawnProjectile bit  within   items/active/weapons/melee/abilities/hammer/fugroundslam

it should have the projectile at the point of impact, and i suppose it does, but the projectile itself spawns to my left, visually, rather than below the player.

**Comment by NotMrFlibble** [comment]
Setting the direction in the world.spawnProjectile call to {1, 0} works, but I'm unsure whether this is the correct fix.

**Comment by renbear67** [comment]
Well, one aspect of this is that the actual location of the impact is between two known locations, but the projectile is being spawned at one of those locations rather than said actual location inbetween.
```JSON
    if world.lineTileCollision(lastSlamPosition, newSlamPosition) then
...
      world.spawnProjectile(self.projectileType, lastSlamPosition, activeItem.ownerEntityId(), {0,0}, false, params)
```
We can probably figure out the actual impact location by using
`List<Vec2I>` world.collisionBlocksAlongLine(`Vec2F` startPoint, `Vec2F` endPoint, [`CollisionSet` collisionKinds], [`int` maxReturnCount])
setting `maxReturnCount` to 1, and then using the first (and only) element of the list returned for the `spawnProjectile()` call.

Now, keep in mind that I haven't tried any of this. Last night, I found the "direction" parameter @NotMrFlibble mentioned, but was still figuring out the location part of it. lastSlamPosition and newSlamPosition might actually be almost (or exactly) the same point, which would render all my speculation moot. I seem to remember @sayterdarkwynd mentioning that the impact happens at the player, though.

**Comment by renbear67** [comment]
Maybe I should have stated outright that I'm hoping for some reactions/opinions from @sayterdarkwynd and/or @NotMrFlibble on this?

**Comment by sayterdarkwynd** [comment]
just try it and see. Im not picky on how its done, only that visually, it makes sense.

**Comment by sayterdarkwynd** [comment]
ill try flibbles fix and see in the meantime

**Comment by sayterdarkwynd** [comment]
so, it does work with that fix...but the majority of the sprite is under theimpactpint /ground 

ill see if tinkeringwith the numbers resolves the location

**Comment by renbear67** [comment]
No, you don't want to hardcode an offset for this. It will differ based on the position of the player and the position of whatever it impacts. That's what I was talking about with the `world.collisionBlocksAlongLine()` stuff. We apparently need to use that to extract the actual point of impact. I just needed to find out if it was working fine without it, and apparently it isn't.

**Comment by renbear67** [comment]
I put in a world.collisionBlocksAlongLine() call, and while it looks better, it's still a little off, I think. The explosion seems to get swallowed by tiles. Still tinkering a bit.

**Comment by renbear67** [comment]
I seem to achieved success. The offset should probably be in the projectile itself, though.

It works for now, anyway. I'll do a commit and pull request.

**Comment by renbear67** [comment]
Hopefully fixed in Pull Request #368
